Assignment 3.Rmd

```diff
@@ -127,29 +128,81 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
 
 ```
 
 What patterns do you see in the plot?
 
-
+There is a two lines run chart in the plot.
 
 It would be useful to determine how many people are in each cluster. We can do this easily with dplyr.
 
 ```{r}
-K7 <- 
+K7 <- dplyr::count(K4, cluster)
+names(K7) <- c("#cluster", "#people")
+
 ```
 
 Look at the number of people in each cluster, now repeat this process for 3 rather than 2 clusters. Which cluster grouping do you think is more informative? Write your answer below:
+```{r}
+fit <- kmeans(K3, 3)
+fit$cluster
+K8 <- data.frame (K3, fit$cluster)
+names(K8) <- c("1","2","3","4","5","cluster")
+K9 <-gather(K8,week,motivation,1:5,na.rm=TRUE,convert=FALSE)
+K10 <- K9 %>% group_by(week,cluster) %>% summarise(avg=mean(motivation))
+K10$week <-as.numeric(K10$week)
+K10$cluster <-as.factor(K10$cluster)
+ggplot(K10,aes(x=week,y=avg,color=cluster))+geom_line()+xlab("Week")+ylab("Average Motivation")
+K11 <- count(K8,cluster)
 
 ##Part II
 
 Using the data collected for Assignment 2 (which classes students were in), cluster the students, then redraw the graph of the class but color the students according the cluster they are in.
+```{r}
+D3 <- read.csv("hudk4050-classes.csv", header = TRUE)
+D4 <- unite(D3,Name,"First.Name","Last.Name",sep = "", remove = TRUE,na.rm = FALSE)
+D5 <- gather(D4, Class, Class.Num,"Class.1","Class.2","Class.3","Class.4","Class.5","Class.6", na.rm = TRUE, convert = FALSE)
+D5$Class.Num <- gsub(pattern = " ",replacement = "",D5$Class.Num)
+D6 <- dplyr::select(D5,Name,Class.Num)
+D6 <- filter(D6,Class.Num>0,Class.Num!="HUDK4050")
+D6$count <- 1
+D6 <- unique(D6)
+D7 <- spread(D6,Class.Num,count,fill=0)
+D7 <- as.data.frame(D7)
+Max1 <- as.matrix(D7)
+row.names(Max1)<- D7$Name
+class(Max1) <- "numeric"
+Max1 <- Max1[,-1]
+Max2 <- Max1%*%t(Max1)
+diag(Max2)<-0
+install.packages("klaR")
+library(klaR)
+fit<-kmodes(Max2,3)
+g <-graph.adjacency(Max2,mode="undirected")
+plot(g,layout=layout.fruchterman.reingold,vertex.color=fit$cluster)
+X <- fit$cluster
+
 
 ##Part III
 
 In class activity 6 you clustered students in the class by the answers to a questionaire. Create a visualization that shows the overlap between these clusters and the clusters generated in part II.
-
+A1<-read.csv("HUDK405019-clustering.csv",header = TRUE)
+A1<-unite(A1,Name, "First.Name","Last.Name",sep = "", remove = TRUE,na.rm = FALSE)
+row.names(A1) <- A1$Name
+A1$Name <- NULL
+A2 <- A1[,-12:-14] 
+A2 <- A2%>% mutate_all(funs(gsub("[a-zA-Z]", "", .)))
+A2 <- A2%>% mutate_all(funs(as.numeric(.)))
+A2 <- as.data.frame(scale(A2))
+A2 <- A2 %>% mutate_all(funs(ifelse(is.na(.) == TRUE, 0, .)))
+fit2 <- kmeans(A2,3) 
+XX <- fit2$cluster
+data.frame(X)
+data.frame(XX)
+library(vcd)
+XXX<-structable(X~XX,direction = NULL, split_vertical = FALSE)
+mosaic(XXX,shade=TRUE, legend=TRUE)
 ## Please render your code as an .html file using knitr and Pull Resquest both your .Rmd file and .html files to the Assignment 3 repository.
```

Comment:
Overall, good job on the coding part of this assignment but you may want to work on its analysis aspect.